common/src/main/java/com/microsoft/identity/common/DataBagUtil.java

```diff
@@ -20,36 +20,46 @@
 // LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
-package com.microsoft.identity.common.java.eststelemetry;
+package com.microsoft.identity.common;
+
+import android.os.Bundle;
 
 import com.microsoft.identity.common.java.interfaces.INameValueStorage;
+import com.microsoft.identity.common.java.util.ported.DataBag;
 
-import java.util.concurrent.ConcurrentHashMap;
+import java.io.Serializable;
+import java.util.Map;
 
 import lombok.NonNull;
 
-public class TelemetryMap implements INameValueStorage<CurrentRequestTelemetry> {
+/**
+ * {@link DataBag} utilities.
+ */
+public class DataBagUtil {
 
-    private final ConcurrentHashMap<String, CurrentRequestTelemetry> telemetryMap = new ConcurrentHashMap<>();
+    /**
+     * Converts a {@link DataBag} to a {@link Bundle}
+     */
+    public static Bundle toBundle(@NonNull final DataBag dataBag) {
+        final Bundle bundle = new Bundle();
 
-    @Override
-    public CurrentRequestTelemetry get(@NonNull String name) {
-        return telemetryMap.get(name);
-    }
+        for (final String key : dataBag.keySet()) {
+            bundle.putSerializable(key, dataBag.get(key));
+        }
 
-    @Override
-    public void put(@NonNull String name, CurrentRequestTelemetry value) {
-        telemetryMap.put(name, value);
+        return bundle;
     }
 
-    @Override
-    public void remove(@NonNull String name) {
-        telemetryMap.remove(name);
-    }
+    /**
+     * Converts a {@link Bundle} to a {@link DataBag}
+     * */
+    public static DataBag fromBundle(@NonNull final Bundle bundle) {
+        final DataBag result = new DataBag();
 
-    @Override
-    public void clear() {
-        telemetryMap.clear();
+        for (final String key : bundle.keySet()) {
+            result.put(key, bundle.getSerializable(key));
```

Comment:
DataBag.put is marked with a NonNull annotation; however it's valid for getSerializeable to return a null.  See: https://developer.android.com/reference/android/os/Bundle#getSerializable(java.lang.String)

Reply:
the value field is nullable.